paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/PaymentMethodRowButton.kt

```diff
@@ -66,26 +80,91 @@ internal fun PaymentMethodRowButton(
                 enabled = isClickable,
                 onClick = onClick
             ),
+        trailingContent = trailingContent,
+        onClick = onClick
     ) {
         Row(
             horizontalArrangement = Arrangement.spacedBy(12.dp),
             verticalAlignment = Alignment.CenterVertically,
         ) {
             RowButtonInnerContent(isEnabled, iconContent, title, subtitle, contentDescription)
 
-            Spacer(modifier = Modifier.weight(1f))
+            if (style !is RowStyle.FlatWithCheckmark) {
+                Spacer(modifier = Modifier.weight(1f))
+            }
 
             if (promoText != null) {
                 PromoBadge(promoText)
             }
 
-            if (trailingContent != null) {
+            if (trailingContent != null && style !is RowStyle.FlatWithCheckmark) {
```

Comment:
We should add a method, rather than doing an instance check. Or add it as a param to the function so the caller can pass it in based on their requirements.